dotcom-rendering/src/components/Discussion.tsx

```diff
@@ -85,6 +85,96 @@ const remapFilters = (
 	} satisfies FilterOptions;
 };
 
+type State = {
+	comments: CommentType[];
+	isClosedForComments: boolean;
+	isExpanded: boolean;
+	commentPage: number;
+	filters: FilterOptions;
+	hashCommentId: number | undefined;
+	totalPages: number;
+	loading: boolean;
+};
+
+const initialState: State = {
+	comments: [],
+	isClosedForComments: false,
+	isExpanded: false,
+	commentPage: 1,
+	filters: initFiltersFromLocalStorage(),
+	hashCommentId: undefined,
+	totalPages: 0,
+	loading: true,
+};
+
+type Action =
+	| {
+			type: 'commentsLoaded';
+			comments: CommentType[];
+			isClosedForComments: boolean;
+			totalPages: number;
+	  }
+	| { type: 'expandComments' }
+	| { type: 'addComment'; comment: CommentType }
+	| { type: 'updateCommentPage'; commentPage: number; shouldExpand: boolean }
+	| { type: 'updateHashCommentId'; hashCommentId: number | undefined }
+	| { type: 'filterChange'; filters: FilterOptions; commentPage?: number }
+	| { type: 'updateTotalPages'; totalPages: number }
+	| { type: 'setLoading'; loading: boolean };
+
+const reducer = (state: State, action: Action): State => {
+	switch (action.type) {
+		case 'commentsLoaded':
+			return {
+				...state,
+				comments: action.comments,
+				isClosedForComments: action.isClosedForComments,
+				totalPages: action.totalPages,
+				loading: false,
+			};
+		case 'addComment':
+			return {
+				...state,
+				comments: [action.comment, ...state.comments.slice(0, -1)], // Remove last item from our local array Replace it with this new comment at the start
+				isExpanded: true,
+			};
+		case 'expandComments':
+			return {
+				...state,
+				isExpanded: true,
+			};
+		case 'updateCommentPage':
+			return {
+				...state,
+				commentPage: action.commentPage,
+				isExpanded: action.shouldExpand ? true : state.isExpanded,
```

Comment:
I cannot picture why we would ever want changing the page not to expand the discussion.

Reply:
When we first load comments we set the comment page in state. In this case, we wouldn't want to expand the comments as it just on initial load.

Reply:
I wondering if this should be its own action, but we can be address as a follow-up